src/lib/containers/DirectDownload.tsx

```diff
@@ -52,27 +52,43 @@ const DirectDownload: React.FunctionComponent<DirectFileDownloadProps> = props =
   }, [accessToken, inView])
 
   const getDownloadLink = async () => {
-    let preSignedURL
-    try {
-      if (associatedObjectType === FileHandleAssociateType.TableEntity) {
-        const files = await getTableEntityFileHandle(true)
-        preSignedURL = files.requestedFiles[0].preSignedURL!
-      } else {
-        const file = await getFileResult(fileEntity!, accessToken, false, true)
-        preSignedURL = file.preSignedURL
+      let preSignedURL
+      // SWC-5907: opening in the file must be strictly done in the same click event process (Safari only).
+      // https://stackoverflow.com/questions/6628949/window-open-popup-getting-blocked-during-click-event
+      const isSafari = navigator.userAgent.indexOf("Safari") != -1 && navigator.userAgent.indexOf("Chrome") == -1
```

Comment:
I figure user agents are pretty stable but I wonder if we would be better served using an dependency here. Or if this should be pulled out and moved to utils

Reply:
That may be a good idea.  I'll look into it.